src/MagnumPlugins/ObjImporter/ObjImporter.cpp

```diff
@@ -28,81 +28,338 @@
+        indices[0] = strToInt(word);
+        return std::make_tuple(indices, endpos);
+    }
+
+    indices[0] = strToInt(pos.prefix(i));
+    endpos = pos.suffix(i+1);
+
+    i = findNext(endpos, ' ', true, true); /* Texture coordinates are not terminated by '/', but ' ' or newline */
+    if(i == -1)
+        i = findNext(endpos, '\r', true, true);
+    if(i == -1)
+        i = findNext(endpos, '\n', true, true);
+
+    if(i != -1) { /* There may not be a texCoord! Eg. "1/" */
+        auto prefix = endpos.prefix(i);
+        if(!prefix.empty()) {
+            indices[1] = strToInt(prefix);
+            endpos = endpos.suffix(i);
+        }
+    }
+
+    return std::make_tuple(indices, endpos);
+}
+
+// Parse a "v/n/t" string to indices
+// Warning: I'm not handing cases like "v/n", where the normal is not terminated by '/', but ' '!
+// Same for "v", where even the normal is omitted.
+std::tuple<std::array<int, 3>, ArrayView<char>> parseVertex(const ArrayView<char> pos) {
+    std::array<int, 3> indices{0, 0, 0};
+    ArrayView<char> endpos{};
+
+    int i = findNext(pos, '/', true, true);
+    if(i == -1) {
+        /* v1 v2 rather than v1/t1 v2/t2 or v1/ v2/ */
+        ArrayView<char> word;
+        std::tie(word, endpos) = nextWord(pos);
+        indices[0] = strToInt(word);
+        return std::make_tuple(indices, endpos);
     }
 
-    Math::Vector<size, Float> output;
+    indices[0] = strToInt(pos.prefix(i));
+    endpos = pos.suffix(i+1);
+
+    i = findNext(endpos, '/', true, true);
+    if(i != -1) { /* there may not be a normal! Eg. "1//2", in which case the indices of the / are 1 apart */
+        auto prefix = endpos.prefix(i);
+        if(!prefix.empty()) {
+            indices[1] = strToInt(prefix);
+        }
+        endpos = endpos.suffix(i+1);
+    }
 
-    for(std::size_t i = 0; i != size; ++i) {
-        #ifndef CORRADE_TARGET_ANDROID
-        output[i] = std::stof(data[i]);
-        #else
-        /* CRAPPY ANDROID GODDAMIT! It's also not exposed in std:: namespace,
-           unlike std::strtoul() and others. WTF! */
-        char* end{};
-        output[i] = strtof(data[i].data(), &end);
-        if(end == data[i].data() || output[i] == HUGE_VALF)
-            throw std::exception{};
-        #endif
+    i = findNext(endpos, ' ', true, true); /* texture coordinates are not terminated by '/', but ' ' or newline */
+    if(i == -1)
+        i = findNext(endpos, '\r', true, true);
+    if(i == -1)
+        i = findNext(endpos, '\n', true, true);
+
+    if(i != -1) { // there may not be a texCoord! Eg. "1//"
+        auto prefix = endpos.prefix(i);
+        if(!prefix.empty()) {
+            indices[2] = strToInt(prefix);
+            endpos = endpos.suffix(i);
+        }
     }
 
-    if(data.size() == size+1) {
-        /* This should be obvious from the first if, but add this just to make
-           Clang Analyzer happy */
-        CORRADE_INTERNAL_ASSERT(extra);
+    return std::make_tuple(indices, endpos);
+}
```

Comment:
I think this could be all made much less complex and less prone to file errors by doing just a single pass over the data instead of first trying to find the delimiter using a highly-parametrized `findNext()` and then trying to parse everything before that as a number, because that way you can very well end up with half of a broken file trying to be parsed as a number ;) Or, on the other hand, index list like `378baaad/1/5` would be parsed as 378, 1, 5 without any error reported because the `baaad` part gets skipped by `findNext()`.

So, the parsing could look like this (very similar to what the ODDL parser is doing):

1. Ignore all whitespace from the front of the line
2. Branch on first non-whitespace character
3. If it's a `f`, for example, this is a face
4. Ignore all whitespace until next character
5. Because this is `f` line, next should be a number. Eat all numeric chars and parse them as a number. If there are none, end with an error.
6. Next character after the parsed number should be either a space or `/`, branch based on that. If it's not, end with an error.
7. Again expect numeric chars next, parse them.
8. After a valid sequence of face indices is parsed, next should be just whitespace until EOL. If not, emit an error (for example -- "too many vertices per face" or "garbage at EOL" etc.).
9. Ignore `\r` or `\n` characters.
10. Start from 1 again.